src/Features/Core/Portable/MakeFieldReadonly/AbstractMakeFieldReadonlyDiagnosticAnalyzer.cs

```diff
@@ -0,0 +1,194 @@
+            {
+                if (item is IFieldSymbol symbol &&
+                    symbol.DeclaredAccessibility == Accessibility.Private &&
+                    !symbol.IsReadOnly &&
+                    !symbol.IsConst &&
+                    !symbol.IsImplicitlyDeclared &&
+                    !IsMutableValueType(symbol.Type))
+                {
+                    var isInCurrentNode = false;
+                    foreach (var syntaxReference in item.DeclaringSyntaxReferences)
+                    {
+                        if (!syntaxReference.SyntaxTree.Equals(context.Node.SyntaxTree))
+                        {
+                            continue;
+                        }
+
+                        if (!context.Node.Span.Contains(syntaxReference.Span))
+                        {
+                            continue;
+                        }
+
+                        isInCurrentNode = true;
+                        break;
+                    }
+
+                    if (isInCurrentNode)
+                    {
+                        nonReadonlyFieldMembers.Add(symbol);
+                    }
+                }
+            }
+
+            var syntaxFactsService = GetSyntaxFactsService();
+            foreach (var syntaxReference in typeSymbol.DeclaringSyntaxReferences)
+            {
+                var typeNode = syntaxReference.SyntaxTree.GetRoot(context.CancellationToken).FindNode(syntaxReference.Span);
+
+                var semanticModelForTree = context.SemanticModel.Compilation.GetSemanticModel(syntaxReference.SyntaxTree);
+                RemoveAssignedSymbols(semanticModelForTree, syntaxFactsService, typeNode, nonReadonlyFieldMembers, context.CancellationToken);
+            }
+
+            foreach (var symbol in nonReadonlyFieldMembers)
+            {
+                var diagnostic = Diagnostic.Create(
+                    GetDescriptorWithSeverity(option.Notification.Value),
+                    symbol.Locations[0]);
+                context.ReportDiagnostic(diagnostic);
+            }
+
+            nonReadonlyFieldMembers.Free();
+        }
+
+        private void RemoveAssignedSymbols(SemanticModel model, ISyntaxFactsService syntaxFactsService, SyntaxNode node, PooledHashSet<IFieldSymbol> unassignedSymbols, CancellationToken cancellationToken)
+        {
+            foreach (var descendant in node.DescendantNodes())
+            {
+                if (unassignedSymbols.Count == 0)
+                {
+                    return;
+                }
+
+                if (!(descendant is TIdentifierNameSyntax name))
+                {
+                    continue;
+                }
+
+                var symbol = model.GetSymbolInfo(descendant).Symbol as IFieldSymbol;
+                if (symbol == null || !unassignedSymbols.Contains(symbol))
+                {
+                    continue;
+                }
+
+                if (!IsMemberOfThisInstance(descendant))
```

Comment:
i don't think this is right.  you might be in a nested type.

Reply:
➡️ Covered by the next comment.